### PR TITLE
BugFix: Last character of SQLSTATE error code is not copied to nanodbc::database_error::state()

### PR DESCRIPTION
**Pull request by beemike5**
## What does this PR do?

1. Bugfix: Last character of ODBC state was not copied to nanodbc::database_error exception, so nanodbc::database_error::state returned only with the first four character of the SQLSTATE.

2. Fix Docker build: docker build failed with the following error:

        **Reading package lists...
        W: The repository 'http://archive.ubuntu.com/ubuntu zesty Release' does not have a Release file.
        W: The repository 'http://archive.ubuntu.com/ubuntu zesty-updates Release' does not have a Release file.
        W: The repository 'http://security.ubuntu.com/ubuntu zesty-security Release' does not have a Release file.
        W: The repository 'http://archive.ubuntu.com/ubuntu zesty-backports Release' does not have a Release file.
        E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/zesty/universe/source/Sources  404  Not Found [IP: 185.125.190.36 80]
        E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/zesty-updates/universe/source/Sources  404  Not Found [IP: 185.125.190.36 80]
        E: Failed to fetch http://security.ubuntu.com/ubuntu/dists/zesty-security/universe/source/Sources  404  Not Found [IP: 91.189.91.38 80]
        E: Failed to fetch http://archive.ubuntu.com/ubuntu/dists/zesty-backports/main/binary-amd64/Packages  404  Not Found [IP: 185.125.190.36 80]
        E: Some index files failed to download. They have been ignored, or old ones used instead.
        The command '/bin/bash -o pipefail -c apt-get -qy update && apt-get -qy install --no-upgrade --no-install-recommends         apt-transport-https         apt-utils         curl         software-properties-common         && apt-get clean         && rm -rf /var/lib/apt/lists/*' returned a non-zero code: 100**
      
      Solution:
      https://askubuntu.com/questions/999856/apt-get-update-fails-when-updating-from-17-04-to-17-10-after-eol-none-of-the-mi

### CODE REVIEW

File: nanodbc/nanodbc.cpp [hunk 1]
```diff
@@ -3743,7 +3743,9 @@ connection::connection(const string& connection_string, long timeout)
 {
 }
 
-connection::~connection() noexcept {}
+connection::~connection() noexcept
+{
+}
```
Comment (mloskot) [inline nanodbc/nanodbc.cpp:3748]: Does this and the rest reformatting come from our `.clang-format`?

File: test/test_case_fixture.h [hunk 2]
```diff
@@ -37,6 +37,29 @@ struct test_case_fixture : public base_test_fixture
...
 
+    template <typename Functor>
+    void check_database_error_state(
+        Functor operationThrowsDatabaseError,
+        const std::string& expectedState)
+    {
+        bool database_error_thrown = false;
+        try
+        {
+            operationThrowsDatabaseError();
```
Comment (mloskot) [inline test/test_case_fixture.h:48]: Should read `operation_throws_database_error`, and `expected_state` too.

File: test/test_case_fixture.h [hunk 3]
```diff
@@ -1904,16 +1932,15 @@ struct test_case_fixture : public base_test_fixture
...
-            NANODBC_TEXT("9.100")};
+        nanodbc::string::value_type trunc_float[batch_size][6] = {NANODBC_TEXT("1.100"),
+                                                                  NANODBC_TEXT("2.300"),
+                                                                  NANODBC_TEXT("3.100"),
+                                                                  NANODBC_TEXT("4.500"),
+                                                                  NANODBC_TEXT("5.700"),
+                                                                  NANODBC_TEXT("6.000"),
+                                                                  NANODBC_TEXT("7.900"),
+                                                                  NANODBC_TEXT("8.900"),
+                                                                  NANODBC_TEXT("9.100")};
```
Comment (mloskot) [inline test/test_case_fixture.h:1943]: Did you use our `.clang-format`?

**Review by mloskot — CHANGES_REQUESTED** [review]
Thanks for the contribution.
Please, see my comments.

The formatting is only allowed if
  1. generated by our `.clang-format`
  2. submitted in separate commits - please do not mix functional and style changes in single commit.

File: nanodbc/nanodbc.cpp [hunk 4]
```diff
@@ -408,14 +408,14 @@ recent_error(SQLHANDLE handle, SQLSMALLINT handle_type, long& native, std::strin
     } while (rc != SQL_NO_DATA);
 
     convert(std::move(result), rvalue);
-    if (size(sql_state) > 0)
+
+    auto stateSize = size(sql_state);
+    if (stateSize > 0)
```
[the remark this answers is not in the document]
Comment (beemike5) [reply nanodbc/nanodbc.cpp]: Thanks, fixed!

File: nanodbc/nanodbc.cpp (hunk 1 above)
Comment (beemike5) [reply nanodbc/nanodbc.cpp]: Used the style script, but from within the container (guess different clang-format version). When I ran the script from the host os, then it changed back these lines.

File: test/test_case_fixture.h (hunk 2 above)
Comment (beemike5) [reply test/test_case_fixture.h]: Thanks, fixed!

File: test/test_case_fixture.h (hunk 3 above)
Comment (beemike5) [reply test/test_case_fixture.h]: Used the style script, but from within the container (guess different clang-format version). When I ran the script from the host os, then it changed back these lines.

**Comment by mloskot** [comment]
I'm closing this PR in favour of PR #298  which offers more complete set of tests and is clearer without any unrelated style changes.

**Pull request closed without merging**